### PR TITLE
Invites: First pass at invites for Jetpack sites

### PR DESCRIPTION
**Pull request by ebinnion**
This change requires that D5206 and a Jetpack patch be deployed first.

Testing instructions to come later.

### CODE REVIEW

**Comment by ebinnion** [comment]
At this point, I've got the PR so that:

- We disable the invite form and show a notice if the site does not have SSO
- We disable the invite form and show a notice if the site does not have Jetpack 4.9+
- The "Add" button in the "People" menu item of the "My Sites" sidebar (that's a mouthful) points to the invite form when the `jetpack/invites` feature flag is `true`.

screen shot 2017-04-21 at 12 39 12 pm
screen shot 2017-04-21 at 12 39 19 pm

File: client/my-sites/people/invite-people/index.jsx [hunk 1]
```diff
@@ -286,113 +293,175 @@ const InvitePeople = React.createClass( {
...
+								{ translate(
+									'(Optional) You can enter a custom message of up to 500 characters ' +
+									'that will be included in the invitation to the user(s).'
+								) }
+							</FormSettingExplanation>
+						</FormFieldset>
+
+						<FormButton disabled={ this.isSubmitDisabled() } onClick={ this.onClickSendInvites } >
+							{ translate(
+								'Send Invitation',
```
Comment (a8ci18n) [inline client/my-sites/people/invite-people/index.jsx]: Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Send Invitation', 'Send Invitations', { context: 'Button label'} )` *ES Score: 21*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/jetpack-invites)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Comment (ebinnion) [inline client/my-sites/people/invite-people/index.jsx]: This is the same translation. Just on multiple lines.

File: client/my-sites/people/invite-people/index.jsx [hunk 2]
```diff
@@ -286,113 +293,175 @@ const InvitePeople = React.createClass( {
...
 
+	renderInviteForm() {
+		const {
+			site,
+			translate,
+			needsVerification,
+			isJetpack,
+			isSSOActive
+		} = this.props;
+		const inviteForm = (
```
Comment (ebinnion) [inline client/my-sites/people/invite-people/index.jsx]: This is being moved from the render method, but should all be the same code.

File: client/my-sites/people/invite-people/index.jsx (hunk 1 above)
Comment (rralian) [inline client/my-sites/people/invite-people/index.jsx]: That's weird. Shouldn't this warning only happen if the string would create a new translation request @yoavf? That doesn't appear to be the case here, as the translation is just coming from another part of the code.

File: client/my-sites/people/invite-people/index.jsx [hunk 3]
```diff
@@ -286,113 +295,183 @@ const InvitePeople = React.createClass( {
...
+			</Card>
+		);
+
+		// Return early for WPCOM or needs verification
+		if ( ! site || ! isJetpack || needsVerification ) {
+			return inviteForm;
+		}
+
+		const jetpackVersion = get( site, 'options.jetpack_version', 0 );
+		if ( versionCompare( jetpackVersion, '4.9-alpha', '<' ) ) {
```
Comment (rralian) [inline client/my-sites/people/invite-people/index.jsx]: I assume that 4.9-alpha will be the version we run on AT sites in order to get this functionality into AT sites early?

Comment (ebinnion) [inline client/my-sites/people/invite-people/index.jsx]: I think we'll actually need to update this to `4.10-alpha` now that we missed the release. But, yes, we'd need to run whatever version includes the new endpoints in Jetpack.

Comment (rralian) [inline client/my-sites/people/invite-people/index.jsx]: Actually... we just launched `4.9`, so shouldn't this target `5.0-alpha` instead? And how will we target the Automated Transfer hosts to launch this functionality early for them? Do we need two `versionCompare()` checks?

File: client/my-sites/people/invite-people/index.jsx [hunk 4]
```diff
@@ -286,113 +295,183 @@ const InvitePeople = React.createClass( {
...
+			return (
+				<div className="invite-people__action-required">
+					<Notice
+						status="is-warning"
+						showDismiss={ false }
+						text={ translate( 'Inviting users requires Jetpack 4.9 or higher' ) }>
```
Comment (rralian) [inline client/my-sites/people/invite-people/index.jsx]: should be `5.0` as I understand (or `4.10`?)

**Comment by rralian** [comment]
Oh, I thought of an issue. Right now this is designed so the invite form is unavailable to Jetpack sites until we set the configuration for `jetpack/invites` to `true`. And then we show it to all Jetpack users for that environment and if they don't have the latest version they are shown a message to upgrade to the latest version of Jetpack.

But we want to expose this functionality to AT sites _prior_ to the feature being released across all of Jetpack. We don't want to expose this to Jetpack users if the version they need to support it hasn't been released yet.

So I think we need some other controls around this. Maybe in every place we check `config.isEnabled( 'jetpack/invites' )` we should do an OR check of `config.isEnabled( 'jetpack/invites' ) || this.props.isAutomatedTransfer` (that's assuming we use the `isAutomatedTransfer` selector).

File: client/my-sites/people/invite-people/index.jsx (hunk 1 above)
Comment (yoavf) [inline client/my-sites/people/invite-people/index.jsx]: @rralian @ebinnion yeah, I'll look into it. https://github.com/Automattic/gp-localci/issues/24

**Comment by rralian** [comment]
:shipit:

**Pull request merged**